GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -110,18 +118,50 @@ public bool TryGetLooseObjectId(string directoryName, string filePath, out strin
         /// Creates a pack file from loose objects
         /// </summary>
         /// <returns>The number of loose objects added to the pack file</returns>
-        public int CreateLooseObjectsPackFile()
+        public bool TryCreateLooseObjectsPackFile(out int objectsAddedToPack)
         {
-            int objectsAddedToPack = 0;
+            int localObjectCount = 0;
 
             GitProcess.Result result = this.RunGitCommand(
                 (process) => process.PackObjects(
                     "from-loose",
                     this.Context.Enlistment.GitObjectsRoot,
-                    (StreamWriter writer) => objectsAddedToPack = this.WriteLooseObjectIds(writer)),
+                    (StreamWriter writer) => localObjectCount = this.WriteLooseObjectIds(writer)),
                 nameof(GitProcess.PackObjects));
 
-            return objectsAddedToPack;
+            if (result.ExitCodeIsSuccess)
+            {
+                objectsAddedToPack = localObjectCount;
+                return true;
+            }
+            else

```

Comment:
Is a corrupt object the only reason why we wouldn't return ExitCodeIsSuccess?

Reply:
Maybe not? I'll scan the error message first.